### PR TITLE
Gcc03

### PR DESCRIPTION
**Pull request by preetithakur**

### CODE REVIEW

**Comment by preetithakur** [comment]
vocado run --test-runner runner gcc.py
Fetching asset from gcc.py:GCC.test

rhel8.6

JOB ID : 6c4e7494ff5c1f77b62f4d7aa68733d417a317e0
JOB LOG : /root/preeti4/avocado-fvt-wrapper/results/job-2022-02-21T08.55-6c4e749/job.log
(1/1) gcc.py:GCC.test: -FAIL: Few gcc tests failed,refer the log file (8050.81 s)
RESULTS : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML : /root/preeti4/avocado-fvt-wrapper/results/job-2022-02-21T08.55-6c4e749/results.html
JOB TIME : 8051.53 s

rhel9
root@ltcever87-lp9 toolchain]# avocado run --test-runner runner gcc.py
Fetching asset from gcc.py:GCC.test
JOB ID : cf80ac86673bb698c8c8f1eb7ad6a1b4c77d0703
JOB LOG : /preeti_new/avocado-fvt-wrapper/results/job-2022-02-23T02.27-cf80ac8/job.log
(1/1) gcc.py:GCC.test: CANCEL: Failed to install guile-devel required for this test. (1.35 s)
RESULTS : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML : /preeti_new/avocado-fvt-wrapper/results/job-2022-02-23T02.27-cf80ac8/results.html
JOB TIME : 14.74 s

tests will not pass, there are failures seen.
so to make test run above package is added
bugzilla for the failures will be raised.

the changes applicable are for both RHEL8.6 and RHEL9

File: toolchain/gcc.py [hunk 1]
```diff
@@ -52,13 +52,12 @@ def setUp(self):
                              'gmp-devel', 'glibc-devel', 'mpfr-devel',
                              'makeinfo', 'texinfo', 'mpc-devel'])
         else:
-            packages.extend(['glibc-static', 'autogen', 'guile',
-                             'guile-devel', 'libgo', 'libgo-devel',
-                             'libgo-static', 'elfutils-devel',
+            packages.extend(['glibc-static', 'guile-devel', 'elfutils-devel',
```
Comment (sacsant) [inline toolchain/gcc.py:55]: This would still fail complaining about guile-devel package.  Move this package to if(rhel8)

**Review by sacsant — CHANGES_REQUESTED** [review]

File: toolchain/gcc.py [hunk 2]
```diff
@@ -52,13 +52,12 @@ def setUp(self):
...
                              'texinfo-tex', 'texinfo', 'elfutils-libelf-devel',
                              'gmp-devel', 'mpfr-devel', 'libmpc-devel',
-                             'gcc-gnat', 'libgnat', 'zlib-devel',
-                             'gettext', 'libgcc', 'libgomp'])
+                             'zlib-devel', 'gettext', 'libgcc', 'libgomp'])
+            if dist.name == 'rhel8':
+                packkages.extend(['autogen', 'guile'])
```
Comment (sacsant) [inline toolchain/gcc.py:60]: There is a type in packkages. Fix this as well.

**Review by sacsant — CHANGES_REQUESTED** [review]

File: toolchain/gcc.py (hunk 2 above)
Comment (preetithakur) [reply toolchain/gcc.py:60]: sure will update

**Comment by PraveenPenguin** [comment]
@preetithakur can you please have a valid commit message aligned with the change proposed

File: toolchain/gcc.py [hunk 3]
```diff
@@ -52,12 +52,12 @@ def setUp(self):
                              'gmp-devel', 'glibc-devel', 'mpfr-devel',
                              'makeinfo', 'texinfo', 'mpc-devel'])
         else:
-            packages.extend(['glibc-static', 'guile-devel', 'elfutils-devel',
+            packages.extend(['glibc-static', 'elfutils-devel',
```
Comment (sacsant) [inline toolchain/gcc.py:55]: This looks like an incremental patch on top of previous changes. Can you include all the changes (previous and current) into a single commit?

**Review by sacsant — CHANGES_REQUESTED** [review]

**Comment by preetithakur** [comment]
closing this PR as raised another PR where issues raised are addressed

**Pull request closed without merging**